wrangler-core/src/main/java/io/cdap/directives/column/SetType.java

```diff
@@ -121,8 +131,38 @@ public Schema getOutputSchema(SchemaResolutionContext context) {
         .map(
           field -> {
             try {
-              return field.getName().equals(col) ?
-                Schema.Field.of(col, ColumnConverter.getSchemaForType(type, scale)) : field;
+              if (field.getName().equals(col)) {
+                String schemaString = field.toString();
```

Comment:
You can use field.getSchema() to get the schema of the field, check if it is of type decimal and getPrecision(), getScale() should work

Reply:
Done